go/vt/topotools/vschema_ddl.go

```diff
@@ -180,6 +180,59 @@ func ApplyVSchemaDDL(ksName string, ks *vschemapb.Keyspace, ddl *sqlparser.DDL)
 			}
 		}
 		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "vindex %s not defined in table %s.%s", name, ksName, tableName)
+
+	case sqlparser.AddVschemaColStr:
+		if table == nil {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "table %s.%s not defined in vschema", ksName, tableName)
+		}
+		for _, column := range table.Columns {
+			if ddl.VschemaCol.Name.EqualString(column.Name) {
+				return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "table %s.%s has column:%s already defined in vschema", ksName, tableName, column.Name)
+			}
+		}
+		table.Columns = append(table.Columns, &vschemapb.Column{Name: ddl.VschemaCol.Name.Lowered(), Type: ddl.VschemaCol.Type.SQLType()})
+		ks.Tables[tableName] = table
+		return ks, nil
+
+	case sqlparser.DropVschemaColStr:
+		if table == nil {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "table %s.%s not defined in vschema", ksName, tableName)
+		}
+		found := false
+		purged := make([]*vschemapb.Column, 0, len(table.Columns))
+		for _, col := range table.Columns {
+			if !ddl.VschemaCol.Name.EqualString(col.Name) {
+				purged = append(purged, col)
+			} else {
+				found = true
+			}
+		}
+		if !found {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "table %s.%s has no column:%s in vschema", ksName, tableName, ddl.VschemaCol.Name.Lowered())
+		}
+		table.Columns = purged
+		ks.Tables[tableName] = table
+		return ks, nil
+
+	case sqlparser.SetVschemaUpdatesStr:
+		name := ddl.Table.Name.String()
+		if _, ok := ks.Tables[name]; !ok {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "vschema does not contain table %s in keyspace %s", name, ksName)
+		}
+		for _, update := range ddl.VschemaUpdates {
+			// for now the only setting we accept is `authoritative` or `column_list_authoritative`
+			if colName := update.Name.Name; colName.EqualString("authoritative") || colName.EqualString("column_list_authoritative") {
```

Comment:
Minor nit — this isn’t a column name so something like `setting` would be clearer than `colName`